src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -122,9 +124,22 @@ private void proxyWmsRequest(HttpServletRequest request, HttpServletResponse res
 
             URL wmsUrl = new URL(wmsUrlStr + "?" + queryString);
 
-            IOUtils.copy(wmsUrl.openConnection().getInputStream(), response.getOutputStream());
+            HttpURLConnection connection = (HttpURLConnection)wmsUrl.openConnection();
+            if (connection.getResponseCode() != 200 ) {
+                String ret = String.format("ERROR proxying URL '%s' - %s", wmsUrl, connection.getResponseMessage());
+                response.sendError(connection.getResponseCode(), ret);
+                LOGGER.log(Level.SEVERE, ret);
+            }
+            else {
+                response.setStatus(connection.getResponseCode());
+                response.setContentType(connection.getContentType());
+                response.setContentLength(connection.getContentLength());
+                IOUtils.copy(connection.getInputStream(), response.getOutputStream());
```

Comment:
I *think* the response stream needs protection against an Exception, because IOUtils.copy() won't close it after finishing the copy action. https://stackoverflow.com/questions/4325576/socket-stream-hangs-when-using-ioutils-copy
 

Using try-with, https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html it would look something like this, 

```Java
try (OutputStream out = response.getOutputStream() ) {
  IOUtils.copy(connection.getInputStream(), response.getOutputStream());
}
```